src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1481,6 +1481,70 @@ public void ListWithOnlyHalfConnectableHeadersPresented_TheFirstNonconsecutiveHe
             currentHeaders[allHeaders[2].HashBlock].Next.Should().BeEmpty();
         }
 
+        /// <summary>
+        /// Issue 39 @ Initial chain is 20 headers long. Single checkpoint is at 1000. Max reorg is 10. Finalized height is 10.
+        /// We receive a chain of 100 headers from peer1. Nothing is marked for download. Peer2 presents a chain which forks
+        /// at 50 and goes to 150. Nothing is marked for download but the chain is accepted.
+        /// </summary>
+        [Fact]
+        public void ChainTHasCheckpointAt1000_MaxReorgIs10_TwoChainsPriorTo1000Presented_NothingIsMarkedForDownload()
+        {
+            // Chain header tree setup. Initial chain has 1 header and it uses checkpoints.
+            // Example: h1=h2=...=h20.
+            const int initialChainSize = 20;
+            TestContext ctx = new TestContextBuilder().WithInitialChain(initialChainSize).UseCheckpoints().Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+
+            // Set checkpoint at height 1000.
+            const int checkpointHeight = 1000;
+            ChainedHeader fakeChainTip = ctx.ExtendAChain(checkpointHeight - initialChainSize, chainTip);
+            var checkpoint = new CheckpointFixture(checkpointHeight, fakeChainTip.Header);
+            ctx.SetupCheckpoints(checkpoint);
+
+            // Setup max reorg of 10.
+            const int maxReorg = 10;
+            ctx.ChainStateMock.Setup(x => x.MaxReorgLength).Returns(maxReorg);
+
+            // Setup finalized block height to 10.
+            ctx.FinalizedBlockMock.Setup(m => m.GetFinalizedBlockHeight()).Returns(10);
+
+            // Extend a chain by 50 headers.
+            // Example: h1=h2=...=h50.
+            const int extensionSize = 30;
+            chainTip = ctx.ExtendAChain(extensionSize, chainTip);
+
+            // Setup chain A that has 100 headers and is based on the previous 30 header extension.
+            // Example: h1=h2=..=h50=a51=a52=..=a120.
+            const int chainAExtensionSize = 70;
+            ChainedHeader chainATip = ctx.ExtendAChain(chainAExtensionSize, chainTip);
+            List<BlockHeader> listOfChainAHeaders =
+                ctx.ChainedHeaderToList(chainATip, extensionSize + chainAExtensionSize);
+
+            // Peer 1 presents a chain A.
+            // Chain accepted but nothing marked for download.
+            const int peer1Id = 1;
+            ConnectNewHeadersResult connectionResult = cht.ConnectNewHeaders(peer1Id, listOfChainAHeaders);
+            connectionResult.DownloadFrom.Should().BeNull();
+            connectionResult.DownloadTo.Should().BeNull();
```

Comment:
As mentioned in different test, if you test download from and download to, please also include the check of block availability using the extension method we have for that.

Same also couple of lines below.

Reply:
Done